tracer/src/Datadog.Trace/RemoteConfigurationManagement/RcmSubscriptionManager.cs

```diff
@@ -156,10 +149,19 @@ public byte[] GetCapabilities()
     {
         // capabilitiesArray needs to be big endian
 #if NETCOREAPP
-        var capabilitiesArray = _capabilities.ToByteArray(true, true);
+        var capabilitiesArray = _capabilities.ToByteArray(isUnsigned: true, isBigEndian: true);
 #else
         var capabilitiesArray = _capabilities.ToByteArray();
         Array.Reverse(capabilitiesArray);
+
+        if (capabilitiesArray.Length > 1 && capabilitiesArray[0] == 0)
```

Comment:
Totally agree that at this point it feels like `BigInteger` is doing more harm than good here... but is it worth just fixing that in this PR rather than punting it down the road?

Reply:
I also agree that we should move away from `BigInteger` and I think it is worth fixing ASAP, but I'd rather it be in a follow up PR than this one IMO just to keep things simple for this PR

Reply:
> I'd rather it be in a follow up PR than this one IMO just to keep things simple for this PR

That, and the fact that I need to finish #5453 asap. I can come back and do this `BitInteger` refactoring after that remote config PR is done.